src/oc_reporter_zipkin.erl

```diff
@@ -66,10 +66,46 @@ zipkin_span(Span, LocalEndpoint) ->
        <<"shared">> => false, %% TODO: get from attributes?
        <<"localEndpoint">> => LocalEndpoint,
        %% <<"remoteEndpoint">> =>  %% TODO: get from attributes?
-       <<"annotations">> => [],
+       <<"annotations">> => to_annotations(Span#span.time_events),
        <<"tags">> => to_tags(Span#span.attributes) %% TODO: merge with oc_tags?
      }.
 
+to_annotations(TimeEvents) ->
+    to_annotations(TimeEvents, []).
+
+to_annotations([], Annotations) ->
+    Annotations;
+to_annotations([{Timestamp, #annotation{description=Description,
+                                        attributes=Attributes}} | Rest], Annotations) ->
+    to_annotations(Rest, [#{<<"timestamp">> => wts:to_absolute(Timestamp),
+                            <<"value">> => annotation_value(Description, Attributes)} | Annotations]);
+to_annotations([{Timestamp, MessageEvent=#message_event{}} | Rest], Annotations) ->
+    to_annotations(Rest, [#{<<"timestamp">> => wts:to_absolute(Timestamp),
+                            <<"value">> => annotation_value(MessageEvent)} | Annotations]).
+
+annotation_value(Description, Attributes) ->
+    AttrString = lists:join(", ", [[Key, "=", to_string(Value)] ||
+                                      {Key, Value} <- maps:to_list(Attributes)]),
+    iolist_to_binary([Description, " Attributes:{", AttrString, "}"]).
+
+annotation_value(#message_event{type=Type,
+                                id=Id,
+                                uncompressed_size=UncompressedSize,
+                                compressed_size=CompressedSize}) ->
+    iolist_to_binary(["MessageEvent:{type=", atom_to_binary(Type, utf8),
+                      ", id=", integer_to_binary(Id),
```

Comment:
integer_to_binary - converts to decimal value or binary? real binary may be hard to read.

Reply:
It is more like integer to string. Erlang has a specific binary type. It is used a lot for strings, esp when going to json since regular Erlang strings are just lists of integers and the way most json encoders work in Erlang is to require strings as binaries so it can easily know if you meant a string or an array of integers.

Binaries are also represented with the `<<` and `>>` syntax you probably see all over the code.